### DISCUSSION TITLE
Stalled list of available tags for proxy cache - any way to force upstream call for update?

### DISCUSSION THREAD

**Discussion by szpak** [goharbor/harbor#18187] [Q&A] [answered]
I use Renovate to bump the artifact & container versions. To reduce network usage and avoid rate limiting, I also use Harbor as a proxy cache to DockerHub (in addition to keep my own images). It works pretty well, but recently, I have bumped into a small issue.

Renovate makes the following call to check the available tags for a given (proxied) image, e.g:
> `https://my-harbor/v2/proxy/library/maven/tags/list?n=10000`

("proxy" is a proxy cache project in Harbor, "library/maven" is an [artifact in DockerHub](https://hub.docker.com/_/maven))

In response, Harbor returns (only) the locally available (previously used) tags. No extra hit to DockerHub is performed to fetch all the tags available upstream. When I explicitly request some other version/tag in some other build, it is fetched by Harbor and that new tag is available in the another Renovate's tags/list call. Effectively, it makes the Harbor proxy cache mechanism not very useful for using with Renovate (to discover new upstream tags).

**Q. Is it possible to configure Harbor to perform upstream call on tags/list requests? Possibly they could be cached for "some (configured) time" to do not hit the rate limiting and to make that calls faster.**

Tested with Harbor v2.1.3-b6de84c5. I know, it is quite old, however, I don't have a newer version available right now. In addition, I haven't seen any related change in the newer changelogs, so I suspect it might be an issue also with the current Harbor version.

**Comment by stonezdj** [comment, marked as answer]
Please check it on Harbor 2.5.0 or later, this pr https://github.com/goharbor/harbor/pull/16394 add this feature.

**Reply by szpak** [reply]
Cool, I will check it out. Thanks for you answer.
